keycloak/extensions/src/main/java/com/github/thomasdarimont/keycloak/custom/endpoints/filter/AuthFilter.java

```diff
@@ -0,0 +1,50 @@
+package com.github.thomasdarimont.keycloak.custom.endpoints.filter;
+
+import com.github.thomasdarimont.keycloak.custom.support.KeycloakSessionLookup;
+import jakarta.ws.rs.ForbiddenException;
+import jakarta.ws.rs.NotAuthorizedException;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerRequestFilter;
+import jakarta.ws.rs.ext.Provider;
+import org.keycloak.authorization.util.Tokens;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.representations.AccessToken;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+/**
+ * Custom request filter for token handling. Executed if a request method is annotated with @FilterBinding
+ */
+@Provider
+@AuthFilterBinding
+public class AuthFilter implements ContainerRequestFilter {
+
+    private final KeycloakSession keycloakSession = KeycloakSessionLookup.currentSession();
+
+    private final Pattern clientPattern = Pattern.compile("acme.*");
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        // For some reason, the Authorization header is missing from the headers stored in keycloakSession
+        String header = requestContext.getHeaderString("Authorization");
+        String token = null;
+
+        if (header != null) {
+            int index = header.indexOf("Bearer ");
+            if (index >= 0) {
+                token = header.substring(index + 7);
+            }
+        }
+
+        AccessToken accessToken = Tokens.getAccessToken(token, keycloakSession);
```

Comment:
```suggestion
        AccessToken accessToken = Tokens.getAccessToken(token, KeycloakSessionLookup.currentSession());
```
inline and remove `keycloakSession` field.